### DISCUSSION TITLE
Unable to login to DT

### DISCUSSION THREAD

**Discussion by nigellh** [DependencyTrack/dependency-track#3483] [Q&A]
Did a complete refresh of my DT instance with the latest containers etc and deleted the DB as it would not migrate.

Came up fine. 
Logged in with admin/admin and changed the pwd
Created a number of teams and IDs
I logged in with mine. All OK, can create projects etc.

My colleague could not with the default password I gave them.

I logged out and tried with their ID and pwd and could get in.

So we cleared cache, tried different browsers, rebooted, restarted DT nothing. 

I could still log in, so I logged out again and I tried with theirs again and still OK, but now I can no longer log in with mine, admin or theirs.

Symptoms are enter the ID and PWD, select login button and absolutely nothing happens. No response whatsoever.

Any suggestions gratefully received.

I have come across this in the past for a smaller server and I have no idea what made it work.

I was working on Firefox, logged in with theirs on FF and then tried the Safari that they are using and that seems to possibly be what has broken me on either browser. (I think).

I do not want to have to rebuild it again, when it might be something simple.

Oh and no log errors either. The login page comes up fine. Just weird!

**Comment by nscuro** [comment]
> absolutely nothing happens. No response whatsoever.

This is indeed weird, but it may indicate a network problem. Are you able to see the login request being sent in your browser's developer tools (*Network* tab), when pressing the login button? And if you do, are you getting a response, and if so, what does it say?

**Comment by nigellh** [comment]
It gets to the server

![image](https://github.com/DependencyTrack/dependency-track/assets/12110404/5bc6288d-394f-4e40-9b1a-0d363693ca8f)

No idea what it means though.

**Reply by nscuro** [reply]
Requests from the frontend to the API server are being blocked by your browser, because the host where the frontend is running (the requests *Origin*) is not listed as acceptable origin by the API server (refer to [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)). 

You can effectively bypass this by allowing any origin. Uncomment or add these to your `docker-compose.yml`:

https://github.com/DependencyTrack/dependency-track/blob/c9c170253f82ef8f774837e2375b07b2cceebd91/src/main/docker/docker-compose.yml#L73-L78

**Comment by nigellh** [comment]
Sorry to bother you, but this is where my limited knowledge hits

![image](https://github.com/DependencyTrack/dependency-track/assets/12110404/8716364f-34ca-49d8-9c8e-17a9867a6d38)

![image](https://github.com/DependencyTrack/dependency-track/assets/12110404/e2af262d-cc7c-4cd1-a289-9947428a8d72)

Gives me 

![image](https://github.com/DependencyTrack/dependency-track/assets/12110404/5f037ec5-d35e-4df7-8739-e73910eab59d)

Not sure if I need to uncomment environment as well?

**Reply by nscuro** [reply]
No worries. You need to pay attention that you're consistent in how you specify environment variables.

Compose allows two notations:

```yaml
environment:
  ENV_VAR_NAME: "value"
```

And:

```yaml
environment:
  - ENV_VAR_NAME=value
```

Make sure you pick one style and stick to it. If you mix it, you'll get errors.

**Comment by nigellh** [comment]
OK

![image](https://github.com/DependencyTrack/dependency-track/assets/12110404/ef902678-5d6d-44a7-942a-861d8f3d86d0)

![image](https://github.com/DependencyTrack/dependency-track/assets/12110404/aea0c0b1-214d-4433-8af5-5daae589850b)

Need to figure the colon as https://docs.dependencytrack.org/getting-started/configuration/

Does not seem to indicate them

**Comment by nigellh** [comment]
OK so....

![image](https://github.com/DependencyTrack/dependency-track/assets/12110404/283445a6-f5fb-46c7-8079-fa176db213b9)

DT Started, but it did not fix it, have I missed something? I note that you said from line 73 above but I did it from line 72, but either way I still have the same problem :-(

**Reply by nscuro** [reply]
Are you getting the same errors as before, or is it different now?

Can you share the entire `docker-compose.yml` please?

**Comment by nigellh** [comment]
I believe so 

![image](https://github.com/DependencyTrack/dependency-track/assets/12110404/ca6991f6-72f6-47d2-a58e-55bfed10d0d3)

[docker-compose.yml.zip](https://github.com/DependencyTrack/dependency-track/files/14361649/docker-compose.yml.zip)

**Comment by nigellh** [comment]
Hi Niklas, 

I might have fixed it and it is so so so annoying.

      # - API_BASE_URL=https://9.20.193.157:8081
      - API_BASE_URL=http://9.20.193.157:8081

Swapped it from https to http as https is not yet supported. 

It is exceptionally weird as it ran for nearly a year with this setting and no issues and the temp server has it like that with no issues.

Bit more testing, but fingers crossed.

**Comment by nigellh** [comment]
That seems to be it. Just in case anyone else needs a cause and solution in very, very layman's terms.

The cors problem appears to be when the system you are sending a call to responds with a response that makes it look like a different system is responding. Many systems do this and it is OK as long as the system is configured to support it.

For us, we had not changed anything and the system had been fine for months and months with the current docker-compose.yml file and then it just stopped dead in allowing people to log in.

The long and the short of it is that Dependency Track does not yet support HTTPS, but we had our server defined as

      - API_BASE_URL=https://9.20.123.456:8081
 
I can only guess that something changed either on the server or DT or browsers that stopped this being allowed.

Changed it to the following and all seems to be OK.

      - API_BASE_URL=http://9.20.123.456:8081

(Base docker-compose.yml has

      - API_BASE_URL=http://localhost:8081

So should have seen it when we changed that)

**Comment by nigellh** [comment]
Nop that does not appear to be it!

**Comment by nigellh** [comment]
Ignore that it was it!

**Comment by Ringostar1982** [comment]
That did not fix my login CORS Problem:

frontend base URL:
API_BASE_URL=http://localhost:8081

And CORS is NOT commented out in the .yml- file:

> services:
>   dtrack-apiserver:
>     image: dependencytrack/apiserver
>     environment:    
>     
>     # Optional Cross-Origin Resource Sharing (CORS) Headers
>      - ALPINE_CORS_ENABLED=true
>      - ALPINE_CORS_ALLOW_ORIGIN=*
>      - ALPINE_CORS_ALLOW_METHODS=GET, POST, PUT, DELETE, OPTIONS
>      - ALPINE_CORS_ALLOW_HEADERS=Origin, Content-Type, Authorization, X-Requested-With, Content-Length, Accept, Origin, X-Api-Key, X-Total-Count, *
>      - ALPINE_CORS_EXPOSE_HEADERS=Origin, Content-Type, Authorization, X-Requested-With, Content-Length, Accept, Origin, X-Api-Key, X-Total-Count
>      - ALPINE_CORS_ALLOW_CREDENTIALS=true
>      - ALPINE_CORS_MAX_AGE=3600


![deeptrack_CORS](https://github.com/DependencyTrack/dependency-track/assets/29915862/5690ffe7-3e12-4c33-a221-1a09764a99b4)
![deeptrack_CORS2](https://github.com/DependencyTrack/dependency-track/assets/29915862/22825871-4802-48f6-a385-67c1ccfea7ed)

**Reply by nigellh** [reply]
Are you trying to log in from the same server that DT is running on or from another server?

**Reply by Ringostar1982** [reply]
I did the setup from dockerfile in my WSL Ubuntu, then go to Edge Browser and type "http://localhost:8080/"

> Downloads the latest Docker Compose file and start
> curl -LO https://dependencytrack.org/docker-compose.yml
> 
> docker-compose up -d

> [+] Running 2/2
>  ✔ Container deeptracktest-dtrack-apiserver-1  Started                                                                                                                                                        0.0s
>  ✔ Container deeptracktest-dtrack-frontend-1   Started

**Reply by nigellh** [reply]
Sorry, I should have been clearer, Is the browser that you are entering the "http://localhost:8080/" in, on the same system that you are running DT on?

**Reply by Ringostar1982** [reply]
Yes its the same system, win10. The deeptrack is running inside my WSL Console by Docker and "outside" in Windows by the Browser(Edge or Chrome) i try to login on "localhost:8080"

**Reply by Ringostar1982** [reply]
Inside the WSL:
curl http://localhost:8081/api/version
outputs:

> {"version":"4.10.1","timestamp":"2023-12-19T10:56:01Z","systemUuid":"670c7a39-2517-48c7-af9b-efc0c9b82116","uuid":"fed5603d-16fd-4bdb-871c-c2a4a80ee303","application":"Dependency-Track","framework":{"name":"Alpine","version":"2.2.4","timestamp":"2023-12-06T15:54:14Z","uuid":"143d8b46-c86e-45d4-b9cf-b823eef5f2a0"}}

**Reply by nigellh** [reply]
OK, not the expert as some of the guys here and not been involved with running it on Windows at all. 

The only thing I can suggest is to replace the http://localhost:8081/ in the docker-compose.yaml file with the actual IP address of the system that you are running it on and see if that gets you in. For example:

http://192.168.0.23:8081/

Restart DT and see if that helps. 

The issue I had originally is that I was running it on one system, trying to access it from another and it had the http://localhost:8081/ in the yaml file. It would show the web page I just could not log in.

Other than that I will leave it to the real DT experts

**Reply by Ringostar1982** [reply]
No problem. That did not help anyway:
![deeptrack_CORS3](https://github.com/DependencyTrack/dependency-track/assets/29915862/7650c680-2ad7-4b75-86c9-589c2c8504fe)

**Reply by nigellh** [reply]
Just had a thought. Have you just upgraded your DT instance? I believe that I had this after an upgrade and I eventually had to wipe everything and restart. It is due to one of the upgrade versions having a requiring a new version of the database that was incompatible with the old one.

In theory there are instructions to export the DB, create the new one and import what was in there, but I could not get it to work so it was easier to start from scratch, read the users and get them to import their SBOMs.

**Reply by Ringostar1982** [reply]
i didnt do an update. Just installed docker and catched the dockerfile to build the 2 containers all from scratch.

**Reply by nigellh** [reply]
OK, I hope you get it resolved soon!

**Comment by prabhushan** [comment]
Hi,

I am getting the same issue.. i tried for past few days.. Could someone please help.. I made sure all the values are correct as per my knowledge

<img width="1262" height="814" alt="image" src="https://github.com/user-attachments/assets/a271751f-bbb0-4562-8efb-1bb4d163b7ee" />

**Reply by prabhushan** [reply]
[docker-compose.yml](https://github.com/user-attachments/files/23220966/docker-compose.yml)
